scripts/process/clean_county_boundaries.R

```diff
@@ -0,0 +1,255 @@
+    sf_simple %>% select(year, state_FIPS, county_FIPS, state_name, county_short, county_long, geometry)
+  })
+  
+  # combine all years using rbind.sf
+  all_shps_simple <- do.call(rbind, decade_shps_simple)
+  
+  # tidy up and return. use underscore_sep in FIPS because some state and county
+  # codes are longer than the standard, so the _ is needed to disambigute
+  rownames(all_shps_simple) <- NULL
+  all_shps_simple <- all_shps_simple %>%
+    mutate(FIPS_U = paste(state_FIPS, county_FIPS, sep='_'))
+  return(all_shps_simple)
+}
+
+consolidate_state_info <- function(all_shps_simple) {
+  dr_states <- dataRetrieval::stateCd %>%
+    transmute(state_FIPS=STATE, state_abbv=STUSAB, state_name=STATE_NAME)
+  states <- all_shps_simple %>%
+    sf::st_set_geometry(NULL) %>%
+    select(state_FIPS, state_name)  %>%
+    full_join(dr_states, by = c("state_FIPS", "state_name")) %>%
+    distinct() %>%
+    group_by(state_FIPS) %>%
+    summarize(
+      state_name = ifelse(length(na.omit(state_name)) == 1, na.omit(state_name), as.character(NA)),
+      state_abbv = ifelse(length(na.omit(state_abbv)) == 1, na.omit(state_abbv), as.character(NA))) %>%
+    filter(state_FIPS %in% unique(all_shps_simple$state_FIPS)) %>%
+    # use same abbreviation for Alaska and Hawaii both before and after they became states
+    mutate(state_abbv = ifelse(is.na(state_abbv), c('025'='AK','155'='HI')[state_FIPS], state_abbv))
+  return(states)
+}
+
+consolidate_county_info <- function(all_shps_simple) {
+  dr_counties <- dataRetrieval::countyCd %>%
+    transmute(
+      FIPS_U=paste(STATE, COUNTY, sep='_'),
+      state_FIPS=STATE,
+      county_FIPS=COUNTY,
+      county_long=COUNTY_NAME)
+  counties <- all_shps_simple %>%
+    sf::st_set_geometry(NULL) %>%
+    select(FIPS_U, state_FIPS, county_FIPS, county_short, county_long) %>%
+    full_join(dr_counties, by=c("FIPS_U", "state_FIPS", "county_FIPS", "county_long")) %>%
+    distinct() %>%
+    group_by(FIPS_U) %>%
+    summarize(
+      state_FIPS = unique(na.omit(state_FIPS)),
+      county_FIPS = unique(county_FIPS),
+      county_short = {
+        cs <- unique(na.omit(county_short))
+        ifelse(length(cs) == 1, cs, as.character(NA))
+      },
+      county_long  = {
+        cl <- unique(na.omit(county_long))
+        ifelse(length(cl) == 1, na.omit(cl), as.character(NA))
+      }) %>%
+    filter(FIPS_U %in% unique(all_shps_simple$FIPS_U)) %>%
+    mutate(
+      county_short = ifelse(!is.na(county_short), county_short,
+                            ifelse(!is.na(county_long), county_long,
+                                   paste('County', county_FIPS))),
+      county_long = ifelse(!is.na(county_long), county_long,
+                           ifelse(!is.na(county_short), county_short,
+                                  paste('County', county_FIPS)))) %>%
+    select(-FIPS_U)
+  return(counties)
+}
+
+consolidate_county_polygons <- function(state_shp) {
+  # check shape validity and fix if needed/possible
```

Comment:
Note I've gotten some mileage out of this "st_make_valid" from this package: https://github.com/r-spatial/lwgeom

Reply:
thanks for the link. i'm hesitant to try that one because it appears to be tough to install on windows, and if we ever end up needing new collaborators to run this code, the extra install headache doesn't seem to be worth it yet. they say `st_buffer(x,0)` works pretty well most of the time, which so far seems to be the case here. https://github.com/r-spatial/sf/issues/487. but yeah, if that ends up being insufficient, good to keep `st_make_valid` in mind.